### PR TITLE
use nanoid for preact / react todomvc

### PR DESCRIPTION
**Pull request by flashdesignory**
This pr resolves: https://github.com/WebKit/Speedometer/issues/204

For comparison - function to remove:

```
function uuid() {
  let uuid = "";
  for (let i = 0; i < 32; i++) {
      const random = (Math.random() * 16) | 0;
      if (i === 8 || i === 12 || i === 16 || i === 20)
          uuid += "-";

      let currentNumber = random;
      if (i === 12)
          currentNumber = 4;
      else if (i === 16)
          currentNumber = 8 | (random & 3);
      uuid += currentNumber.toString(16);
  }
  return uuid;
}
```

function to replace with:
```
// Borrowed from https://github.com/ai/nanoid/blob/3.0.2/non-secure/index.js
// This alphabet uses `A-Za-z0-9_-` symbols. A genetic algorithm helped
// optimize the gzip compression for this alphabet.
const urlAlphabet =
  'ModuleSymbhasOwnPr-0123456789ABCDEFGHNRVfgctiUvz_KqYTJkLxpZXIjQW'

/**
 *
 * @public
 */
function nanoid(size = 21){
  let id = ''
  // A compact alternative for `for (var i = 0; i < step; i++)`.
  let i = size
  while (i--) {
    // `| 0` is more compact and faster than `Math.floor()`.
    id += urlAlphabet[(Math.random() * 64) | 0]
  }
  return id
}
```
@kara

### CODE REVIEW

**Review by julienw — APPROVED** [review]
Looks good to me, thanks for the change!

File: resources/todomvc/architecture-examples/preact/src/app/model.js [hunk 1]
```diff
@@ -1,13 +1,21 @@
-function uuid() {
-    let uuid = "";
-    for (let i = 0; i < 32; i++) {
-        let random = (Math.random() * 16) | 0;
-        if (i === 8 || i === 12 || i === 16 || i === 20)
-            uuid += "-";
+// Borrowed from https://github.com/ai/nanoid/blob/3.0.2/non-secure/index.js
```
Comment (rniwa) [inline resources/todomvc/architecture-examples/preact/src/app/model.js:1]: We should probably include the LICENSE file with this.

File: resources/todomvc/architecture-examples/preact/src/app/model.js [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
+// This alphabet uses `A-Za-z0-9_-` symbols. A genetic algorithm helped
+// optimize the gzip compression for this alphabet.
+const urlAlphabet = "ModuleSymbhasOwnPr-0123456789ABCDEFGHNRVfgctiUvz_KqYTJkLxpZXIjQW";
```
Comment (camillobruni) [inline resources/todomvc/architecture-examples/preact/src/app/model.js:4]: The alphabet here probably doesn't matter all that much for us, but any reason we don't use the latest version 3.X ?
https://github.com/ai/nanoid/blob/3.3.5/non-secure/index.js

File: resources/todomvc/architecture-examples/preact/src/app/model.js (hunk 1 above)
Comment (flashdesignory) [reply resources/todomvc/architecture-examples/preact/src/app/model.js]: added on top of the file.

File: resources/todomvc/architecture-examples/preact/src/app/model.js (hunk 2 above)
Comment (flashdesignory) [reply resources/todomvc/architecture-examples/preact/src/app/model.js]: updated

**Review by rniwa — APPROVED** [review]

**Pull request merged**